extensions/control-plane/provision/provision-aws-s3/src/main/java/org/eclipse/edc/connector/provision/aws/s3/S3ConsumerResourceDefinitionGenerator.java

```diff
@@ -30,18 +30,16 @@ public class S3ConsumerResourceDefinitionGenerator implements ConsumerResourceDe
 
     @Override
     public ResourceDefinition generate(DataRequest dataRequest, Policy policy) {
```

Comment:
This code should not be removed

Reply:
What's the purpose of this? `dataDestination` cannot never be null, and I don't see a good reason to have a default region value (in this case `US_EAST_1`), I think it's something we inherited from the past and it needed to be fixed accordingly.

Reply:
I was referring to the note about the need to integrate the policy engine and the change to existing behavior that may impact users. I would prefer we keep a default region (does it really need to be removed?). However, we decide to change the existing behavior, it should be noted in the PR and also in the release notes.

Reply:
The problem is that the current behavior was to set the default region if the `dataDestination` was null, and it always is, so what's the point? Could be good to modify this logic in a more reasonable: "if bucket is not specified on the data destination, then set a default one? wdyt @jimmarino

Reply:
@ndr-brt yes, that makes more sense. IMO it should be done in a follow-on PR. Also, please keep the comments about policy.

Reply:
@jimmarino the previous condition related to the S3ConsumerResourceDefinitionGenerator is back again but since I think the discussion is related to the case when someone does not specify a region, I opted to refine the condition with `getDataDestination().getProperty(S3BucketSchema.REGION)==null` instead of `if(getDataDestination()=null)` and I made a unit test that evaluates the case when a person does not add bucket name as property (returning NullPointerExcp). 

I think it's obvious and already spoken but I will write it here. The current validators inside `DataRequest` do not allow to create an object with `dataDestination` null. I can return to the original condition but that would mean, if I keep the  `getDataDestination()=null` it would not be possible to test that case